conductor/__beta__/job/job.py

```diff
@@ -0,0 +1,144 @@
+import logging
+import os
+
+import conductor.lib.conductor_submit as conductor_submit
+import conductor.lib.package_utils
+import conductor.lib.api_client
+
+LOG = logging.getLogger(__name__)
+
+class JobError(Exception):
+    pass
+
+class Job(object):
+    
+    def __init__(self):
+    
+        self.upload_paths = []
+        self.software_packages_ids = []
+        self.owner = 'bob'
+        self.priority = 5
+        self.location = ""
+        self.instance_type = "n1-standard-8"
+        self.metadata = {}
+        self.local_upload = True
+        self.auto_retry_policy = {}
+        self.preemptible = True
+        self.chunk_size = 1
+        self.project = "default"
+        self.output_path = ""
+        self.job_title = ""
+        self.docker_image = ""
+        self._dependencies = None
+        self.environment = {}
+        
+        self._dependency_scan_enabled = True
+        self.conductor_job_id = None
+                
+    def _get_task_data(self):
+        pass
+    
+    def _get_frame_range(self):
+        pass
+    
+    def _get_environment(self):
+        
+        packages = [package for package in conductor.lib.api_client.request_software_packages() if package["package_id"] in self.software_packages_ids]
+        return conductor.lib.package_utils.merge_package_environments(packages, base_env=self.environment)
+    
+    def get_output_path(self):
+        return self.output_path
+    
+    def scan_for_dependencies(self):
+        return []
+    
+    def get_dependencies(self):
+        
+        if self._dependencies is None and self._dependency_scan_enabled:            
+            self._dependencies = self.scan_for_dependencies()
+            
+        return self._dependencies + self.upload_paths
+
+    def submit_job(self):

```

Comment:
can we name this method, simply `submit` ?  

`Job.submit()` vs `Job.submit_job()`